src/Types/ArrayType.php

```diff
@@ -29,14 +30,14 @@ public function getKeyType(): ?Type
 	/**
 	 * @param string|int $key
 	 */
-	public function addInvalidPair($key, ?Type $keyType, ?Type $itemType): void
+	public function addInvalidPair($key, ?WithTypeAndValue $keyTypeAndValue, ?WithTypeAndValue $itemTypeAndValue): void
 	{
-		if ($keyType === null && $itemType === null) {
+		if ($keyTypeAndValue === null && $itemTypeAndValue === null) {
 			throw InvalidArgument::create()
 				->withMessage('At least one of key type and item type of invalid pair should not be null');
 		}
 
-		$this->invalidPairs[$key] = [$keyType, $itemType];
+		$this->invalidPairs[$key] = [$keyTypeAndValue !== null ? $keyTypeAndValue->getInvalidType() : null, $itemTypeAndValue !== null ? $itemTypeAndValue->getInvalidType() : null];
```

Comment:
`WithTypeAndValue` should be stored in invalidPairs instead of getting `Type` here